pkg/kube/client.go

```diff
@@ -550,14 +550,44 @@ func perform(infos Result, fn ResourceActorFunc) error {
 		return ErrNoObjectsVisited
 	}
 
-	for _, info := range infos {
-		if err := fn(info); err != nil {
+	errs := make(chan error)
+	go batchPerform(infos, fn, errs)
+
+	for range infos {
+		err := <-errs
+		if err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
+func batchPerform(infos Result, fn ResourceActorFunc, errs chan<- error) {
+	if len(infos) == 0 {
+		return
+	}
+
+	finished := make(chan bool, 10000)
+	kind := infos[0].Object.GetObjectKind().GroupVersionKind().Kind
+	counter := 0
+	for _, info := range infos {
+		currentKind := info.Object.GetObjectKind().GroupVersionKind().Kind
+		if kind != currentKind {
+			// Wait until the previous kind has finished
+			for i := 0; i < counter; i++ {
+				<-finished
+			}
```

Comment:
This logic should be changed to use `sync.WaitGroup` instead of using a channel and the counter